pkg/config/configobserver/insighgtsdatagather_observer.go

```diff
@@ -77,10 +76,5 @@ func (i *insightsDataGatherController) GatherDisabled() bool {
 	i.lock.Lock()
 	defer i.lock.Unlock()
 
-	if slices.Contains(i.gatherConfig.DisabledGatherers, v1alpha1.DisabledGatherer("all")) ||
-		slices.Contains(i.gatherConfig.DisabledGatherers, v1alpha1.DisabledGatherer("ALL")) {
-		return true
-	}
-
-	return false
+	return i.gatherConfig.Gatherers.Mode == v1alpha2.GatheringModeNone
```

Comment:
This is a bit confusing (semantically). I see that disabling all gatherers equals to gather nothing, so `GatheringModeNone` makes total sense. No complaints here.

I think, though, this could bring some doubts in the future about the wording (and previously having checked the `DisabledGatherers`).

Let's introduce some one-line comment about it, what do you think?